src/main.rs

```diff
@@ -422,7 +422,8 @@ async fn forward_request(
             // Canisters don't have to provide certified variables
             (None, None) => true,
         };
-        if !body_valid {
+
+        if !body_valid && !cfg!(feature = "skip_body_verification") {
```

Comment:
So if body is not certified and we do verification we return a 500?
Otherwise continue on. Aha.

Suggestion to put a comment that says that or possibly make a function that has a name to that effect...

Reply:
Or extract the whole certification stuff into another function

Reply:
The current baseline is to return a `500`. This flag makes it so we will not return `500` even if the body doesn't match.

Reply:
Extracting dealing with certification into a function would reduce the complexity and make this more readable.